File: Minimum Steps Required[  04-April-2023 ]/Solution.cpp
//	C++ CODE <<

int minSteps(string str) {
    int acount = 0;
    int bcount = 0;
    char prev = '#';

    for(int i = 0; i < str.length(); i++) {
        char ch = str[i];
        if(prev == ch) continue;
        
        if(ch == 'a') acount++;
        else bcount++;
        
        prev = ch;
    }

    return min(acount, bcount) + 1;
}

File: Minimum Steps Required[  04-April-2023 ]/Solution.java
	>> JAVA CODE <<

	int minSteps(String str) {
	
	    int acount=0;
	    int bcount=0;
	    char prev='#';
	    
	    for(int i=0;i<str.length();i++)
	    {
	        char ch=str.charAt(i);
	        if(prev==ch) continue;
	        
	        if(ch=='a') acount++;
	        else bcount++;
	        
	        prev=ch;
	    }
		
		    return Math.min(acount,bcount)+1;
	}
